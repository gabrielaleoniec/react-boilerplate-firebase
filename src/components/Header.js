import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { startLogOut } from '../actions/auth';

export const Header = ({ startLogOut }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__container">
                <nav>
                    <Link to="/dashboard" className="header__link">
                        <h1 className="header__title">Boilerplate React App</h1>
                    </Link>
                </nav>
                <button onClick={startLogOut} className="button button--link">Logout</button>
            </div>
        </div>

    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogOut: () => dispatch(startLogOut())
})

export default connect(undefined, mapDispatchToProps)(Header);