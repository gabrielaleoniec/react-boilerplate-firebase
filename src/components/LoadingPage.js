import React from 'react';

const LoadingPage = () => {
    return <div className="loading">
        <img src="/img/loader.gif" alt="Loading" className="loading__image" />
    </div>
}

export default LoadingPage;