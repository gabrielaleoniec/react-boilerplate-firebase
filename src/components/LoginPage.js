import React from 'react';
import { connect } from 'react-redux';
import { startLogIn } from '../actions/auth'

export const LoginPage = ({ startLogIn }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Boilerplate React App</h1>
                <p>Tag line for app</p>
                <button onClick={startLogIn} className="button">Login with Google</button>
            </div>
        </div>);
};

const mapDispatchToProps = (dispatch) => ({
    startLogIn: () => dispatch(startLogIn())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);
