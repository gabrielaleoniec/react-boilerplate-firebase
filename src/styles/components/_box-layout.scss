.box-layout {
    align-items: center;
    background: url('/img/gdansk.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background-color: fade-out(white, .15);
    border-radius: 3px;
    padding: $m-size;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    color: $blue;
    line-height: 1;
    margin: 0 0 $s-size;
}