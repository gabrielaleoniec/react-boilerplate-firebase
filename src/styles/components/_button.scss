.button {
    background-color: $vivid-blue;
    border: none;
    color: $white;
    cursor: pointer;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $xs-size $s-size;
    text-decoration: none;
    transition: background-color ease-in-out .4;

    &:hover {
        background-color: darken($vivid-blue, 10%);
    }

    &--link {
        background-color: transparent;
        display: inline;

        &:hover {
            background-color: transparent;
            text-decoration: underline;
        }
    }

    &--remove {
        background-color: $red;
        
    }
}
