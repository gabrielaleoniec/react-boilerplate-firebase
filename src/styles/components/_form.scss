.form {
    display: flex;
    flex-direction: column;
    align-self: start;

    &>* {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin-bottom: $s-size;
    font-size: italic;
}