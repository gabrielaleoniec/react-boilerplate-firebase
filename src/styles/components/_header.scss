.header {
    background-color: $steel-blue;
}

.header__title {
    color: $white;
}

.header__link {
    color: $white;
    text-decoration: none;

    h1 {
        margin: 0;
    }
}

.header__container {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
}