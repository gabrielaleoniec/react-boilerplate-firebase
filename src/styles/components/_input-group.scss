.input-group {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: stretch;
    margin-bottom: $s-size;
    max-width: 100vw;
    padding: $s-size 0;
}

.input-group__item {
    display: block;
    margin-bottom: $s-size;

    &>input,
    &>textarea,
    &>select {
        width: 316px;
    }
}

@media(min-width: $desktop-breakpoint) {
    .input-group {
        flex-direction: row;
        flex-wrap: wrap;
        margin-bottom: $l-size;
    }

    .input-group__item {
        margin-right: $s-size;
        margin-bottom: 0;

        &:last-child {
            margin-right: 0;
        }

        &>input,
        &>textarea,
        &>select {
            width: auto;
        }
    }
}