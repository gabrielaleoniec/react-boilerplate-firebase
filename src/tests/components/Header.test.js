import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('Should render Header correctly', () => {
    const wrapper = shallow(<Header startLogOut={() => {}} />);
    expect(wrapper.find('h1').text()).toBe('Boilerplate React App');
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
    const startLogOut = jest.fn();
    const wrapper = shallow(<Header startLogOut={startLogOut} />);
    wrapper.find('button').simulate('click');
    expect(startLogOut).toHaveBeenCalled()
});
