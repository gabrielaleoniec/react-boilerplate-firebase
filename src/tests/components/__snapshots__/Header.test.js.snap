// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__container"
    >
      <nav>
        <Link
          className="header__link"
          to="/dashboard"
        >
          <h1
            className="header__title"
          >
            Boilerplate React App
          </h1>
        </Link>
      </nav>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
